tests/xtimer_mutex_lock_timeout/main.c

```diff
@@ -42,9 +49,38 @@ static const shell_command_t shell_commands[] = {
       cmd_test_xtimer_mutex_lock_timeout_long_unlocked, },
     { "mutex_timeout_long_locked", "locked mutex with long timeout",
       cmd_test_xtimer_mutex_lock_timeout_long_locked, },
+    { "mutex_timeout_long_locked_low",
+      "locked mutex from lower prio thread function with long timeout",
+      cmd_test_xtimer_mutex_lock_timeout_low_prio_thread, },
     { NULL, NULL, NULL }
 };
 
+/**
+ * @brief   thread function and stack for
+ *          cmd_test_xtimer_mutex_lock_timeout_low_prio_thread
+ */
+
+char t_stack[THREAD_STACKSIZE_MAIN];
+
+void *thread_low_prio_test(void *arg)
+{
+    mutex_t *test_mutex = (mutex_t *)arg;
+    msg_t msg;
+
+    puts("THREAD low prio: start");
+
+    mutex_lock(test_mutex);
+    thread_wakeup(main_thread_pid);
+    while (msg_try_receive(&msg) == -1) {}
+
+    mutex_unlock(test_mutex);
+    (void)irq_disable();
+    puts("THREAD low prio: exiting low");
+    msg_send_int(&msg, main_thread_pid);
```

Comment:
why `msg_send_int`? This is clearly not called from interrupt context

Reply:
to join threads. Only way to wait for a thread to finish that has a lower prio as far as I know(is done this way in posix reaper implementation)

Reply:
`msg_send_int` does not yield but still unblocks the other thread.

Reply:
I will split it in a function with documentation

Reply:
I added a function for it.